gridstatus/spp.py

```diff
@@ -193,6 +205,146 @@ def get_load_forecast(self, date, forecast_type="MID_TERM", verbose=False):
 
         return current_day_forecast
 
+    def get_solar_and_wind_forecast_short_term(self, date, end=None, verbose=False):
+        """
+        Returns solar and wind generation forecast for +4 hours in 5 minute intervals.
+        Include actuals for past day in 5 minute intervals.
+
+        Arguments:
+            date (pd.Timestamp|str): date to get data for. Supports "latest" and "today"
+            verbose (bool): print info
+
+        Returns:
+            pd.DataFrame: forecast as dataframe.
+        """
+        # The short_term forecast is delayed up to 2 minutes.
+        buffer_minutes = 2
+
+        if date == "latest":
+            date = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
+        df = self._retrieve_solar_and_wind_forecast_short_term(date, end, verbose)
+
+        df["Forecast Type"] = "SHORT_TERM"
+
+        return df
+
+    @support_date_range("5_MIN")
+    def _retrieve_solar_and_wind_forecast_short_term(
+        self,
+        date,
+        end=None,
+        verbose=False,
+    ):
+        """Solar and wind forecast for +4 hours by 5 minute interval."""
+        # Files do not exist in the future
+        if date > self.now():
```

Comment:
Is this amount of custom logic alright? It allows us to support "today" without a unch of unnecessary requests.